### DISCUSSION TITLE
Disable Button on Click

### DISCUSSION THREAD

**Discussion by mathewhealthi** [Budibase/budibase#8851] [Help] [answered]
We have a button set up that triggers an API call when clicked. We run the risk that if a user clicks the button twice, the API call is triggered twice which would be a problem for us. 

Is there a way to disable a button on the first click so as to ensure a user doesn't click on it more than once?

**Comment by andz-bb** [comment, marked as answer]
Hey @mathewhealthi 

In this case I would suggest using a state variable called `submitted` for example by adding an Update State action on the button to set `submitted` to true. Then in your button's conditions you can set it to disable the button if `State.submitted` is true.

![Screenshot 2022-11-29 at 11 05 16](https://user-images.githubusercontent.com/110921612/204513156-cf4afd37-9bf6-447c-8774-4e61dd4cc316.png)

Hope this helps, if you have any issues with this let me know and I'll be happy to look at it further 😊

**Comment by mathewhealthi** [comment]
@andz-bb thanks, this worked!
